src/components/MTableSummaryRow/index.js

```diff
@@ -2,19 +2,75 @@ import * as React from 'react';
 import withStyles from '@material-ui/core/styles/withStyles';
 import { TableRow, TableCell } from '@material-ui/core';
 import { getStyle } from '../MTableCell/utils';
+import * as CommonValues from '../../utils/common-values';
 import PropTypes from 'prop-types';
 
 export function MTableSummaryRow({
   data,
   columns,
   currentData,
+  rowProps,
   renderSummaryRow
 }) {
   if (!renderSummaryRow) {
     return null;
   }
+
+  function renderPlaceholderColumn(key, numIcons = 1) {
+    const size = CommonValues.elementSize(rowProps);
+    const width = numIcons * CommonValues.baseIconSize(rowProps);
+    return (
+      <TableCell
+        key={`placeholder.${key}`}
+        size={size}
+        padding="none"
+        style={{
+          width: width,
+          padding: '0px 5px',
+          boxSizing: 'border-box'
+        }}
+      />
+    );
+  }
+  const placeholderLeftColumns = [];
+  const placeholderRightColumns = [];
+  let placeholderKey = 0;
+
+  // Create empty columns corresponding to selection, actions, detail panel, and tree data icons
+  if (rowProps.options.selection) {
+    placeholderLeftColumns.push(renderPlaceholderColumn(placeholderKey++));
+  }
+  if (
+    rowProps.actions &&
+    rowProps.actions.filter(
+      (a) => a.position === 'row' || typeof a === 'function'
+    ).length > 0
+  ) {
+    const numRowActions = CommonValues.rowActions(rowProps).length;
+    if (rowProps.options.actionsColumnIndex === -1) {
+      placeholderRightColumns.push(
+        renderPlaceholderColumn(placeholderKey++, numRowActions)
+      );
+    } else if (rowProps.options.actionsColumnIndex >= 0) {
+      placeholderLeftColumns.push(
+        renderPlaceholderColumn(placeholderKey++, numRowActions)
+      );
+    }
+  }
+  if (rowProps.detailPanel && rowProps.options.showDetailPanelIcon) {
+    if (rowProps.options.detailPanelColumnAlignment === 'right') {
+      placeholderRightColumns.push(renderPlaceholderColumn(placeholderKey++));
+    } else {
+      placeholderLeftColumns.push(renderPlaceholderColumn(placeholderKey++));
+    }
+  }
+  if (rowProps.isTreeData) {
+    placeholderLeftColumns.push(renderPlaceholderColumn(placeholderKey++));
+  }
+
```

Comment:
I think I see what you're doing here. If I understand this correctly, this solves the issue where sometimes the columns are not aligned correctly with it's header? Should we put this in a different PR, if so?

Reply:
Yes that's what it fixes.

I would've preferred to separate these into their own PR's, but Github makes that difficult. I would've needed to create separate branches for each one. We needed all these fixes for our product last week and there was no movement on merging PRs.

Reply:
@Llolow I thought this would fix that issue where the body doesn't line up with the headers when editing, but it doesn't look like I was right. That's def something that needs to be fixed though.